plugins/base/src/main/kotlin/transformers/documentables/DefaultDocumentableMerger.kt

```diff
@@ -1,200 +1,324 @@
 package org.jetbrains.dokka.base.transformers.documentables
 
+import org.jetbrains.dokka.DokkaConfiguration
+import org.jetbrains.dokka.DokkaSourceSetID
 import org.jetbrains.dokka.model.*
+import org.jetbrains.dokka.model.properties.ExtraProperty
+import org.jetbrains.dokka.model.properties.MergeStrategy
+import org.jetbrains.dokka.model.properties.WithExtraProperties
 import org.jetbrains.dokka.model.properties.mergeExtras
 import org.jetbrains.dokka.plugability.DokkaContext
 import org.jetbrains.dokka.transformers.documentation.DocumentableMerger
 import org.jetbrains.kotlin.utils.addToStdlib.firstNotNullResult
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
 
-internal object DefaultDocumentableMerger : DocumentableMerger {
-
-    override fun invoke(modules: Collection<DModule>, context: DokkaContext): DModule {
+internal class DefaultDocumentableMerger(val context: DokkaContext) : DocumentableMerger {
+    private val dependencyInfo = context.getDependencyInfo()
 
+    override fun invoke(modules: Collection<DModule>): DModule {
         val projectName =
             modules.fold(modules.first().name) { acc, module -> acc.commonPrefixWith(module.name) }
                 .takeIf { it.isNotEmpty() }
                 ?: "project"
 
-        return modules.reduce { left, right ->
+        return topologicalSort(modules).reduce { left, right ->
```

Comment:
Can you think of some name, that is more "sound"?

Reply:
`topologicalSort` is exact sound name. Can chagne to something like `sortModulesTopologically`, but since it is nested private function, I just called it simply

Reply:
Oh, I think you're right. The name is fine ☺️ 
Sorry!